File: src/Moq/Moq/contentFiles/cs/netstandard2.0/Mocks/Mock.cs
using System;
using System.CodeDom.Compiler;
using System.Reflection;
using System.Runtime.CompilerServices;
using Moq.Sdk;

namespace Moq
{
    /// <summary>
    /// Instantiates mocks for the specified types.
    /// </summary>
    [GeneratedCode("Moq", "5.0")]
    [CompilerGenerated]
    partial class Mock
    {
        /// <summary>
        /// Gets the configuration and introspection for the given mocked instance.
        /// </summary>
        public static IMoq<T> Get<T>(T instance) => new Moq<T>(instance.AsMock());

        /// <summary>
        /// Creates the mock instance by using the specified types to 
        /// lookup the mock type in the assembly defining this class.
        /// </summary>
        private static T Create<T>(MockBehavior behavior, object[] constructorArgs, params Type[] interfaces)
        {
            var mocked = (IMocked)MockFactory.Default.CreateMock(typeof(Mock).GetTypeInfo().Assembly, typeof(T), interfaces, constructorArgs);

            mocked.Initialize(behavior);

            return (T)mocked;
        }
   }
}

File: src/Moq/Moq/contentFiles/vb/netstandard2.0/Mocks/Mock.vb
Option Strict On

Imports System
Imports System.CodeDom.Compiler
Imports System.Reflection
Imports System.Runtime.CompilerServices
Imports Moq.Sdk

Namespace Global.Moq

    <GeneratedCode("Moq", "5.0")>
    <CompilerGenerated>
    Partial Friend Class Mock

        Private Shared Function Create(Of T)(ByVal behavior As MockBehavior, ByVal constructorArgs As Object(), ParamArray interfaces As Type()) As T
            Dim mocked = DirectCast(MockFactory.[Default].CreateMock(GetType(Mock).GetTypeInfo().Assembly, GetType(T), interfaces, constructorArgs), IMocked)

            mocked.Initialize(behavior)

            Return DirectCast(mocked, T)
        End Function

    End Class

End Namespace